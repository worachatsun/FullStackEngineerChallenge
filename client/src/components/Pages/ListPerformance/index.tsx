import React, { FunctionComponent } from "react";
import { FaAngleRight } from "react-icons/fa";
import useSWR from "swr";
import { REVIEWED_LIST_API } from "../../../constants/routes";
import { IAnswerModel, IPerformanceModel } from "../../../interfaces/model";
import Empty from "../../commons/Empty";
import { fetcher } from "../../commons/utils/client";
import Layout from "../../Layout";
import { AnswerContainer, Lable, Table, Tr, TrText } from "./ListPerformance.styled";

const Performance: FunctionComponent = () => {
    const token = localStorage.getItem("token");
    const { data = [] } = useSWR<IPerformanceModel[]>([REVIEWED_LIST_API, token], fetcher);

    return data ? (
        <Layout>
            <Lable>Performance reviews</Lable>
            <Table>
                {data.map((review: IPerformanceModel, index: number) => (
                    <Tr key={index}>
                        <TrText>
                            {review.reviewBy}
                            <FaAngleRight />
                            {review.reviewTo}
                        </TrText>
                        <AnswerContainer>
                            {review.answers.map((answer: IAnswerModel) => (
                                <div key={answer.id}>
                                    {`Question${answer.questionId}: `}
                                    {answer.answer}
                                </div>
                            ))}
                        </AnswerContainer>
                    </Tr>
                ))}
                {data.length === 0 ? <Empty title="No review yet. Please come back later." /> : <div />}
            </Table>
        </Layout>
    ) : (
        <div>Loading</div>
    );
};

export default Performance;
