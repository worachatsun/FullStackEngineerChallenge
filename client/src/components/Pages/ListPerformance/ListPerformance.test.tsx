import React from "react";
import renderer from "react-test-renderer";
import ListPerformance from "./index";

describe("<ListPerformance />", () => {
    test("should render correctly", () => {
        const render = renderer.create(<ListPerformance />).toJSON();
        expect(render).toMatchSnapshot();
    });
});
