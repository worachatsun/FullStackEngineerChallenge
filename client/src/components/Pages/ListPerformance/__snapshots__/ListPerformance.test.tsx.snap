// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListPerformance /> should render correctly 1`] = `
Array [
  <div
    className="sc-AxirZ fqNgBu"
  >
    <div
      className="sc-AxiKw jJjpKK"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="img_logo.png"
      />
    </div>
    <div
      className="sc-AxhCb cupTmd"
    >
      <div
        className="sc-AxhUy iYMzkw"
      >
        
      </div>
      <button
        className="sc-AxjAm fbqdBC"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>,
  <div
    className="sc-AxheI jdaMWS"
  >
    <div
      className="sc-AxgMl kikqEL"
    >
      <div
        className="sc-fzozJi cnJWyh"
      >
        Performance reviews
      </div>
      <div
        className="sc-fzoLsD kLfYbT"
      />
    </div>
  </div>,
]
`;
